src/tree/updater_quantile_hist.h

```diff
@@ -442,7 +442,7 @@ class QuantileHistMaker: public TreeUpdater {
     // list of nodes whose histograms would be built explicitly.
     std::vector<ExpandEntry> nodes_for_explicit_hist_build_;
 
-    enum DataLayout { kDenseDataZeroBased, kDenseDataOneBased, kSparseData };
+    enum class DataLayout { kDenseDataZeroBased, kDenseDataOneBased, kSparseData };
```

Comment:
This line caused compilation to fail. Why was `DataLayout` changed to an enum class? Does it address a compiler warning?

Reply:
Yes, it was producing an enum type unscoped warning. I changed the lines where I saw it was used like this: 

https://github.com/dmlc/xgboost/pull/6197/files/b8b00f57582b9e98e28a0506d58011810cbe1762#diff-10aac201acca00858b390ff0d679434fL1239-R1239

Reply:
Got it. Can you fix the failing build for `testxgboost` (C++ tests with Google Test framework)? You can build it by following instructions in https://xgboost.readthedocs.io/en/latest/contrib/unit_tests.html#running-gtest.

Reply:
@sgavil How it is going? Do you need some help?

Reply:
@hcho3 Well, I had to do some refactoring in the names of the unit tests (tbh not sure if I'm able to do that) from example from `ASSERT_EQ(this->data_layout_, RealImpl::kSparseData);` to `ASSERT_EQ(this->data_layout_, RealImpl::DataLayout::kSparseData);` but that belongs to [dmlc-core](https://github.com/dmlc/dmlc-core) and I'm not sure how to proceed with that.